feat: Issue #1 - Add ability to remove profile picture

PrayagCodes opened the pull request:
## Description
This PR adds a "Remove" button to account settings, which allows users to delete their profile picture and revert to the default avatar.
## Before
- video: https://cap.link/ej1nemxp5xsr0xw
<img width="1915" height="943" alt="image" src="https://github.com/user-attachments/assets/3196d55e-25c1-4bb4-9f03-c0989c2f5d42" />

## After
- video: https://cap.link/a5rs4dz1gd8w43w
<img width="1446" height="829" alt="image" src="https://github.com/user-attachments/assets/08316858-12e3-40ef-8af2-a7d74b42f6d6" />

## Implementation
- video: https://cap.link/y9sqzwhn72wq3e3
<img width="1900" height="938" alt="image" src="https://github.com/user-attachments/assets/0b1052ae-6cd4-4f15-8c8b-0b38db36d8dd" />

## Related Issue
Closes [Issue #1 - Add ability to remove profile picture](https://github.com/codepath/puter/issues/5) #5 
## Changes Made
### Core Functionality
- Added remove button UI in account settings (visible only when user has a profile picture)
- Added Confirmation Model  to prompt the user and confirm the action
- Implemented a click handler to remove the picture and update the UI in real-time
- Modified `update_profile()` helper function to support field deletion
- Button uses danger styling (`button-danger` class) for consistency

### Internationalization
- Added `remove_profile_picture`  / `confirm_remove_profile_picture` translation key to all 38 supported languages
- Refined translations for proper verb forms and grammar in 10 languages

## Implementation Details
- Profile deletion: passes `{picture: null}` to `update_profile()`
- Deletes `picture` field from `/{username}/Public/.profile` JSON file
- UI updates without page refresh
- Button visibility tied to profile picture existence

## Testing Performed
- [x] Button appears when profile picture exists
- [x] Clicking button removes picture and reverts to default avatar
- [x] Button hides after picture removal
- [x] Button reappears when new picture uploaded
- [x] Toolbar profile image updates correctly
- [x] Changes persist after page refresh
- [x] All 38 language translations verified

## Files Modified
- `src/gui/src/helpers.js` - Profile deletion logic
- `src/gui/src/UI/Settings/UITabAccount.js` - UI components and event handlers
- `src/gui/src/i18n/translations/*.js` - All 38 translation files

PrayagCodes committed:
feat: add remove profile picture functionality-#issue-1

- Add 'Remove' button to account settings UI below profile picture
- Button only appears when user has an uploaded profile picture
- Clicking button removes picture and reverts to default avatar
- Update update_profile() helper to handle profile field deletion
- Style button with button-danger class for consistent UI

Changes:
* UITabAccount.js: Add remove button UI, click handler, and show/hide logic
* helpers.js: Update update_profile() to delete fields when value is null

The button is styled as a danger action (red background, white text)
and disappears after removing the picture or when no picture exists.

PrayagCodes committed:
feat: add i18n translation keys-#issue-1

- Add translations for all 38 supported languages

PrayagCodes committed:
feat: Add confirmation dialog for profile picture removal-#issue-1

- Add UIAlert confirmation dialog when user attempts to remove profile picture
- Implement flag to prevent multiple simultaneous removal operations
- Add backdrop overlay to block interactions during confirmation
- Change remove button style from 'primary' to 'danger' for better UX
- Add 'confirm_remove_profile_picture' translation key to all 37 language files

Changes:
- src/gui/src/UI/Settings/UITabAccount.js:
  * Import UIAlert component
  * Add is_removing_profile_picture flag to prevent race conditions
  * Wrap removal logic in try-finally block for proper cleanup
  * Add confirmation dialog with backdrop and danger-styled button

- src/gui/src/i18n/translations/*.js:
  * Add confirmation message in 37 languages
  * Message: 'Are you sure you want to remove your profile picture?'

israx commented on src/gui/src/helpers.js:
```diff
@@ -2633,9 +2633,14 @@ window.update_profile = function(username, key_vals){
-                profile[key] = key_vals[key];
-                // update window.user.profile
-                window.user.profile[key] = key_vals[key];
+                if (key_vals[key] === null || key_vals[key] === undefined || key_vals[key] === '') {
```
Nit: `null` , `undefined` and `''` are falsy values, which means we can substitute code above by


```suggestion
                if (!!key_vals[key]) {
```

PrayagCodes replied:
resolved

israx commented on src/gui/src/UI/Settings/UITabAccount.js:
```diff
@@ -38,6 +39,9 @@ export default {
             h += `<div class="profile-picture change-profile-picture" style="background-image: url('${html_encode(window.user?.profile?.picture ?? window.icons['profile.svg'])}');">`;
             h += `</div>`;
         h += `</div>`;
+        h += `<div style="text-align: center; margin-bottom: 20px;">`;
```
Nit:

Let's make sure we are being consistent with styles here. We should establish a boundary when to use inline-styles vs css. 

inline-styles can be beneficial for referring to javascript code such as the profile picture online 39.

israx commented on src/gui/src/UI/Settings/UITabAccount.js:
```diff
@@ -150,6 +157,56 @@ export default {
 
+        $el_window.find('.remove-profile-picture').on('click', async function (e) {
+            // Prevent multiple simultaneous operations
+            if(is_removing_profile_picture)
```
Nit:

Ideally, instead of deduping this call, we should disable the `remove profile picture` button while the picture is being deleted

PrayagCodes replied:
resolved

israx reviewed (approved):
Overall looks good to me. Great job Prayag. PR description is pretty descriptive. Thank you.

PrayagCodes committed:
fix: simplify falsy checks and disable button during profile removal

- Use !key_vals[key] instead of explicit null/undefined/'' checks
- Disable remove profile picture button during operation instead of using flag
- Enable button when showing after new upload

Addresses reviewer nits
- https://github.com/codepath/puter/pull/22/files/b9cf1b0ef36d0cf5484f4ad495220d7f28d553c4#r2448352449
- https://github.com/codepath/puter/pull/22/files/b9cf1b0ef36d0cf5484f4ad495220d7f28d553c4#r2448377901
